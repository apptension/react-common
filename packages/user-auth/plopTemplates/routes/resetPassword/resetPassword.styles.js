import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
`;

export const TextField = styled.input`
  width: 100%;
`;

export const Form = styled.form`
  display: flex;
  flex-direction: column;
`;

export const ButtonGroup = styled.div`
  text-align: right;
`;
