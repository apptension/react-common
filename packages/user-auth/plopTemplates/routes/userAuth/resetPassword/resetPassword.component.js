import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form/immutable';
import { FormattedMessage } from 'react-intl';

import {
  Container,
  Header,
  Form,
  ButtonGroup,
} from '../../../components/forms/form.styles';
import messages from './resetPassword.messages';

import { TextField } from '../../../components/forms/textField';

export class ResetPassword extends PureComponent {
  static propTypes = {
    intl: PropTypes.object.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    resetPassword: PropTypes.func.isRequired,
    location: PropTypes.object.isRequired,
  };

  handleSubmit = (values) => new Promise(() => {
    const { location } = this.props;

    this.props.resetPassword(
      values.get('password'),
      location.state.user,
      location.state.token,
    );
  });

  render() {
    const { intl, handleSubmit } = this.props;

    return (
      <Container>
        <Header>
          <FormattedMessage {...messages.title} />
        </Header>
        <Form onSubmit={handleSubmit(this.handleSubmit)} noValidate>
          <Field
            component={TextField}
            name="password"
            type="password"
            label={messages.passwordLabel}
            intl={intl}
            messages={messages}
          />

          <ButtonGroup>
            <button type="submit">
              <FormattedMessage {...messages.submit} />
            </button>
          </ButtonGroup>
        </Form>
      </Container>
    );
  }
}
