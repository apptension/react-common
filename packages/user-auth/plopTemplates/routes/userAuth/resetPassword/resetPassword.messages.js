import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'resetPassword.title',
    defaultMessage: 'Reset Password',
  },
  passwordLabel: {
    id: 'resetPassword.fields.password.label',
    defaultMessage: 'Password',
  },
  passwordPresenceError: {
    id: 'resetPassword.fields.password.error.presence',
    defaultMessage: 'Password cannot be blank',
  },
  passwordLengthTooShortError: {
    id: 'resetPassword.fields.password.error.lengthTooShort',
    defaultMessage: 'Password should have at least 8 characters',
  },
  passwordLengthTooLongError: {
    id: 'resetPassword.fields.password.error.lengthTooLong',
    defaultMessage: 'Password should be no longer than 32 characters',
  },
  passwordTooCommonError: {
    id: 'resetPassword.fields.password.error.tooCommon',
    defaultMessage: 'Password has been found in a common passwords database. Please, choose a different one.',
  },
  passwordIsTooSimilarToUsernameError: {
    id: 'resetPassword.fields.password.error.tooSimilar',
    defaultMessage: 'This password is too similar to your username. Please, pick a different password.',
  },
  submit: {
    id: 'resetPassword.submit',
    defaultMessage: 'Reset Password',
  },
  cancel: {
    id: 'resetPassword.back',
    defaultMessage: 'Cancel',
  },
});
