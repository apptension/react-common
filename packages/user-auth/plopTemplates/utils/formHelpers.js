import reportError from 'report-error';
import { List } from "immutable";
import { when, is, pipe } from 'ramda';

export const getFormFieldHelperText = (intl, messages, { error, invalid, touched }) => {
  if (invalid && touched) {
    try {
      return intl.formatMessage(messages[pipe(
        when(
          is(List),
          (errors) => errors.first()
        ),
        when(
          is(Array),
          ([firstError]) => firstError
        ),
      )(error)]);
    } catch (e) {
      reportError(e);
    }
  }
  return '';
};
