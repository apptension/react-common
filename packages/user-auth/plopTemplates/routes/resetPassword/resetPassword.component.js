import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form/immutable';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';

import { getFormFieldHelperText } from '../../utils/formHelpers';
import { Container, TextField, Form, ButtonGroup } from './resetPassword.styles';
import messages from './resetPassword.messages';

export class ResetPassword extends PureComponent {
  static propTypes = {
    intl: PropTypes.object.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    resetPassword: PropTypes.func.isRequired,
    location: PropTypes.object.isRequired,
  };

  handleSubmit = (values) => new Promise(() => {
    const { location } = this.props;

    this.props.resetPassword(
      values.get('password'),
      location.state.user,
      location.state.token,
    );
  });

  renderTextField = ({ meta, input, type, label }) => {
    return (
      <TextField
        error={meta.invalid && meta.touched}
        value={input.value}
        type={type}
        label={label}
        helperText={getFormFieldHelperText(this.props.intl, messages, meta)}
        InputProps={input}
      />
    );
  };

  render() {
    const { intl, handleSubmit } = this.props;

    return (
      <Container>
        <FormattedMessage {...messages.title} />
        <Form onSubmit={handleSubmit(this.handleSubmit)} noValidate>
          <Field
            component={this.renderTextField}
            name="password"
            type="password"
            label={intl.formatMessage(messages.passwordLabel)}
          />

          <ButtonGroup>
            <button type="submit">
              <FormattedMessage {...messages.submit} />
            </button>
          </ButtonGroup>
        </Form>
      </Container>
    );
  }
}
