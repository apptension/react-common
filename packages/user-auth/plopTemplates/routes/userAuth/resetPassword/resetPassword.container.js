import { reduxForm } from 'redux-form/immutable';
import validate from 'validate.js';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'ramda';
import { injectIntl } from 'react-intl';

import { RESET_PASSWORD_FORM, UserAuthActions } from '../../../modules/userAuth';
import { ResetPassword } from './resetPassword.component';

const mapStateToProps = createStructuredSelector({});

export const mapDispatchToProps = (dispatch) => bindActionCreators({
  resetPassword: UserAuthActions.resetPassword,
}, dispatch);

export default compose(
  injectIntl,
  connect(mapStateToProps, mapDispatchToProps),
  reduxForm({
    form: RESET_PASSWORD_FORM,
    validate: (values) => validate(values.toJS(), {
      password: {
        presence: {
          message: '^passwordPresenceError',
        },
        length: {
          minimum: 8,
          maximum: 32,
          tooShort: '^passwordLengthTooShortError',
          tooLong: '^passwordLengthTooLongError',
        },
      },
    }),
  }),
)(ResetPassword);
