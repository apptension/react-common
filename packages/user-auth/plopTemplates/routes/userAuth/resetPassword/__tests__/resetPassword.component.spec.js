import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import { expect } from 'chai';
import { Map } from 'immutable';
import { identity } from 'ramda';

import { ResetPassword } from '../resetPassword.component';
import { Form } from '../../../../components/forms/form.styles';
import { intlMock } from '../../../../utils/testUtils';

describe('<ResetPassword />', () => {
  const defaultProps = {
    intl: intlMock(),
    handleSubmit: identity,
    resetPassword: identity,
    location: {
      state: {
        user: 'fakeUid',
        token: 'fakeToken',
      },
    },
  };

  const component = (props) => shallow(
    <ResetPassword {...defaultProps} {...props} />
  );

  it('should render itself', () => {
    const wrapper = component();
    global.expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should call handleSubmit when form is submit', () => {
    const handleSubmit = spy();
    const wrapper = component({ handleSubmit });
    wrapper.find(Form).simulate('submit');

    expect(handleSubmit).to.have.been.called;
  });

  it('should call resetPassword prop when handling submit', () => {
    const password = 'test123';
    const { location: { state: { user, token } } } = defaultProps;
    const handleSubmit = spy();
    const resetPassword = spy();

    const wrapper = component({ handleSubmit, resetPassword });
    wrapper.find(Form).simulate('submit');

    const [fn] = handleSubmit.getCall(0).args;

    fn(Map({ password }));

    expect(resetPassword).to.have.been.calledWith(password, user, token);
  });
});
