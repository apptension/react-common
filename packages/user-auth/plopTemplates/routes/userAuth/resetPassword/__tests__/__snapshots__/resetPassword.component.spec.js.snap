// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPassword /> should render itself 1`] = `
.c0 {
  width: 100%;
  max-width: 600px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: auto;
}

<div
  className="c0"
>
  <styled.h1>
    <FormattedMessage
      defaultMessage="Reset Password"
      id="resetPassword.title"
      values={Object {}}
    />
  </styled.h1>
  <styled.form
    noValidate={true}
    onSubmit={[Function]}
  >
    <Field
      component={[Function]}
      intl={
        Object {
          "formatMessage": [Function],
        }
      }
      label={
        Object {
          "defaultMessage": "Password",
          "id": "resetPassword.fields.password.label",
        }
      }
      messages={
        Object {
          "cancel": Object {
            "defaultMessage": "Cancel",
            "id": "resetPassword.back",
          },
          "passwordIsTooSimilarToUsernameError": Object {
            "defaultMessage": "This password is too similar to your username. Please, pick a different password.",
            "id": "resetPassword.fields.password.error.tooSimilar",
          },
          "passwordLabel": Object {
            "defaultMessage": "Password",
            "id": "resetPassword.fields.password.label",
          },
          "passwordLengthTooLongError": Object {
            "defaultMessage": "Password should be no longer than 32 characters",
            "id": "resetPassword.fields.password.error.lengthTooLong",
          },
          "passwordLengthTooShortError": Object {
            "defaultMessage": "Password should have at least 8 characters",
            "id": "resetPassword.fields.password.error.lengthTooShort",
          },
          "passwordPresenceError": Object {
            "defaultMessage": "Password cannot be blank",
            "id": "resetPassword.fields.password.error.presence",
          },
          "passwordTooCommonError": Object {
            "defaultMessage": "Password has been found in a common passwords database. Please, choose a different one.",
            "id": "resetPassword.fields.password.error.tooCommon",
          },
          "submit": Object {
            "defaultMessage": "Reset Password",
            "id": "resetPassword.submit",
          },
          "title": Object {
            "defaultMessage": "Reset Password",
            "id": "resetPassword.title",
          },
        }
      }
      name="password"
      type="password"
    />
    <styled.div>
      <button
        type="submit"
      >
        <FormattedMessage
          defaultMessage="Reset Password"
          id="resetPassword.submit"
          values={Object {}}
        />
      </button>
    </styled.div>
  </styled.form>
</div>
`;
